### DISCUSSION TITLE
How to delay Alpine init?

### DISCUSSION THREAD

**Discussion by Lomacar** [alpinejs/alpine#2241] [1. Help] [answered]
I want to run a bunch of preparatory code before Alpine does it's thing. The preparation involves async stuff. So how can/should I delay Alpine init until everything is ready? I know there is `Alpine.start()` but I suppose running that more than once could cause unpredictable behaviour. (I am trying to create a standalone library that "plays nice" with Alpine.) Do I have to use modules?

UPDATE:
OK, apparently what I thought was happening wasn't happening and I stumbled into a solution without needing this question answered. But maybe I will leave it here out of curiosity.

**Comment by SimoTod** [comment, marked as answer]
the npm version doesn't autostart so you would manually call `Alpine.start()` whenever you are ready. (https://alpinejs.dev/essentials/installation#as-a-module)
You can't delay the CDN version any further than deferring the script

**Reply by Lomacar** [reply]
OK, thanks.
